Kernel/Bus/VirtIO/Device.cpp

```diff
@@ -113,15 +109,13 @@ bool Device::setup_queues(u16 requested_queue_count)
     }
 
     dbgln_if(VIRTIO_DEBUG, "{}: Setting up {} queues", m_class_name, m_queue_count);
-    for (u16 i = 0; i < m_queue_count; i++) {
-        if (!setup_queue(i))
-            return false;
-    }
-    for (u16 i = 0; i < m_queue_count; i++) { // Queues can only be activated *after* all others queues were also configured
-        if (!m_transport_entity->activate_queue({}, i))
-            return false;
-    }
-    return true;
+    for (u16 i = 0; i < m_queue_count; i++)
+        TRY(setup_queue(i));
+
+    // NOTE: Queues can only be activated *after* all others queues were also configured
+    for (u16 i = 0; i < m_queue_count; i++)
+        TRY(m_transport_entity->activate_queue({}, i));
```

Comment:
Not going to block this PR but I am curious: What happens if we activate 14 out of 16 queues, for example, and then return with an error? 

Is the system in an inconsistent state? I think this kind of "lack of strong exception guarantee" (for lack of a better word) is a general problem in our TRY-heavy code.

Reply:
For VirtIO devices, failing to initialize the device results in a flow where host software should indicate this explicitly by assigning failure status to the status bits, iirc.